### PR TITLE
Add TypeScript type definition

### LINKED ISSUE #60: Typescript support
Calendar needs index.d.ts file or @types so project with typescript base can have their type checking

### PR DESCRIPTION
**Pull request by thebrodmann**
@Kiarash-Z @arminyahya Let's work on this PR to resolve #60.

### CODE REVIEW

**Comment by thebrodmann** [comment]
Also related to #63 and #64.

File: package.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
   "version": "1.0.1",
   "description": "A modern, beautiful, customizable date picker for React",
   "main": "lib/index.js",
+  "types": "index.d.ts",
```
Comment (arminyahya) [inline package.json]: What is reason to add types to package.json, I tested worked fine without that.

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+
+export type Day = {
+  year: number;
+  month: number;
+  day: number;
+};
+
+export type Range<T> = { from?: T; to?: T };
```
Comment (arminyahya) [inline index.d.ts]: Why Range is generic? I see its Day everywhere.

File: package.json (hunk 1 above)
Comment (thebrodmann) [reply package.json]: I don't have any idea why it works but as mentioned in [TypeScript documentation](https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html) we should add it.

File: index.d.ts (hunk 2 above)
Comment (thebrodmann) [reply index.d.ts]: I agree. There is no other use case for it.

**Comment by Kiarash-Z** [comment]
@thebrodmann is there a need for providing default values in here? 🤔(like `'0.4s'` for `slideAnimationDuration` or `#0eca2d` for `colorPrimary)

**Comment by thebrodmann** [comment]
@Kiarash-Z we cannot define default values for properties in the **type** definition file (as those are **value**) but it's important to make such properties optional.

**Comment by Kiarash-Z** [comment]
@thebrodmann will they work fine this way? I mean will default values apply correctly?

**Comment by thebrodmann** [comment]
> will they work fine this way? I mean will default values apply correctly?

Yes. The functionality of the library is JavaScript's business, even in TypeScript projects. Type definition just explains to the TypeScript compiler which type a value should be. (e.g. If a value was defined to be a `string`, passing non-`string` value will cause TypeScript compiler to complain about it.)

**Comment by Kiarash-Z** [comment]
@thebrodmann Alright. We're good to go then 🔥

**Pull request merged**